py-geopolars/python/geopolars/internals/geoseries.py

```diff
@@ -205,6 +205,29 @@ def geoarrow_coords_to_numpy(struct_array: pyarrow.StructArray):
 
         raise ValueError()
 
+    @classmethod
+    def points_from_xy(cls, x, y,
```

Comment:
I think you also want the type hints on `x` and `y`? Note that `Iterable | pl.Series` is a strict subset of what you can pass into `pyarrow.StructArray`; it also accepts numpy arrays and pyarrow arrays I think